File: Arrays/contain_duplicate.py
class Solution:
    def containsDuplicate(self, nums):
        # import pdb; pdb.set_trace()  # Adding the debugger here
        hashset = set()

        for n in nums:
            if n in hashset:
                return True
            hashset.add(n)
        return False

# Example usage
sol = Solution()
print(sol.containsDuplicate([1, 2, 3, 1]))  # Expected output: True
print(sol.containsDuplicate([1, 2, 3, 4]))  # Expected output: False
print(sol.containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]))  # Expected output: True

File: golang/Array/contain_duplicate.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {
	if len(nums) <= 1 {
		return false
	}

	xm := make(map[int]struct{})

	for _, v := range nums {
		if _, ok := xm[v]; ok {
			return true
		}

		xm[v] = struct{}{}
	}

	return false
}

func main() {
	nums := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 1}
	fmt.Println(containsDuplicate(nums)) // Output: true
}